src/poseidon/septidon/params.rs

```diff
@@ -1,41 +1,62 @@
-use super::super::primitives::{Mds as MdsT, P128Pow5T3Compact, Spec};
-use lazy_static::lazy_static;
+use super::super::primitives::{Mds as MdsT, P128Pow5T3Constants};
 
 /// This implementation can be limited to gate degree 5. However, this mode will not work with
 /// blinding or inactive rows. Enable only with a prover that supports assignments to all n rows.
 pub const GATE_DEGREE_5: bool = false;
 
-/// This implementation supports only the scalar field of BN254 at the moment.
-///
-/// To implement for the Pasta curves, adjust the parameters below, and replace the transition round
-/// by a copy, to get 56 rounds instead of 57.
-pub use halo2_proofs::halo2curves::bn256::Fr as F;
+/// This is the base "hashable" type requirement for septidon
+pub trait CachedConstants: P128Pow5T3Constants {
+    /// cached round constants
+    fn cached_round_constants() -> &'static [[Self; 3]];
+    /// cached mds
+    fn cached_mds() -> &'static Mds<Self>;
+    /// cached inversed mds
+    fn cached_mds_inv() -> &'static Mds<Self>;
+}
 
 pub mod sbox {
     use super::super::util::pow_5;
-    use super::F;
+    use halo2_proofs::arithmetic::FieldExt;
```

Comment:
The code does not compile at the moment.

We changed some things in halo2 recently to use the trait `PrimeField` instead of `FieldExt`, that might be the problem. See https://github.com/scroll-tech/poseidon-circuit/pull/13

![compile error](https://user-images.githubusercontent.com/8718243/230478900-d45926b7-6a56-4609-9eba-0e6c85eaa5f6.png)

Reply:
That is cause by the halo2 lib depency: the `scroll-dev-1220` use another poseidon lib without a tag so it points to some unstable code and issue raised after several updating in the codebase.

I had update halo2 to `scroll-dev-0220` (which is consistent with current zkevm-circuit) and the issue has been resolved.